src/orchestrator-process/src/lib.rs

```diff
@@ -716,6 +716,14 @@ async fn spawn_process(
 
     impl Drop for KillOnDropChild {
         fn drop(&mut self) {
+            if let Some(pid) = self.0.id().and_then(|id| i32::try_from(id).ok()) {
+                let _ = nix::sys::signal::kill(
+                    nix::unistd::Pid::from_raw(pid),
+                    nix::sys::signal::Signal::SIGTERM,
+                );
+                // Give the process a bit of time to react to the signal
+                std::thread::sleep(Duration::from_millis(500));
```

Comment:
This is going to synchronously block on an async thread, which is not great. I think a `tokio::task::block_in_place` is in order.

Reply:
Agree+thanks, TIL :)